include/sys/__assert.h

```diff
@@ -22,7 +22,7 @@
 #endif
 
 #if defined(CONFIG_ASSERT_VERBOSE)
-#define __ASSERT_PRINT(fmt, ...) printk(fmt, ##__VA_ARGS__)
+#define __ASSERT_PRINT(fmt, ...) assert_print(fmt, ##__VA_ARGS__)
 #else /* CONFIG_ASSERT_VERBOSE */
 #define __ASSERT_PRINT(fmt, ...)
 #endif /* CONFIG_ASSERT_VERBOSE */
```

Comment:
This will cause some link errors because `CONFIG_ASSERT_VERBOSE` can be set without `CONFIG_ASSERT` (it even defaults to `y`).

A quick fix would be to do `defined(CONFIG_ASSERT) && defined(CONFIG_ASSERT_VERBOSE)`, but a more fundamental and proper fix would be to make `config ASSERT_VERBOSE` and all the ASSERT-related configs depend on `config ASSERT` (i.e. put them inside `if ASSERT`).